src/field_theory/ratfunc.lean

```diff
@@ -458,17 +469,102 @@ variables {K}
 
 section lift_hom
 
-variables {G₀ L : Type*} [comm_group_with_zero G₀] [field L]
+variables {G₀ L R F : Type*} [comm_group_with_zero G₀] [field L] [comm_ring R]
+
+/-- Lift a monoid homomorphism that maps polynomials `φ : polynomial K →* polynomial R`
+to a `ratfunc K →* ratfunc R`,
+on the condition that `φ` maps non zero divisors to non zero divisors,
+by mapping both the numerator and denominator and quotienting them. -/
+def map [monoid_hom_class F (polynomial K) (polynomial R)] (φ : F)
```

Comment:
Is `K` a field or a just a ring? The notation suggests a field. I would expect `K` and `R` to be on equal footing. So either `K`-and-`L` or `R`-and-`S`.

Reply:
In the rest of the file, `K` is often bouncing between just `[comm_ring]`, or `[comm_ring] [is_domain]` or `[field]`. For these lift definitions, I provide variable names that are more ring-like instead.